src/hii_road.py

```diff
@@ -3,356 +3,258 @@
-            .add(highway_escape.multiply(self.highway_escape_weighting))
-            .add(highway_footway.multiply(self.highway_footway_weighting))
-            .add(highway_living_street.multiply(self.highway_living_street_weighting))
-            .add(highway_mini_roundabout.multiply(self.highway_mini_roundabout_weighting))
-            .add(highway_motorway.multiply(self.highway_motorway_weighting))
-            .add(highway_motorway_link.multiply(self.highway_motorway_link_weighting))
-            .add(highway_path.multiply(self.highway_path_weighting))
-            .add(highway_pedestrian.multiply(self.highway_pedestrian_weighting))
-            .add(highway_primary.multiply(self.highway_primary_weighting))
-            .add(highway_primary_link.multiply(self.highway_primary_link_weighting))
-            .add(highway_raceway.multiply(self.highway_raceway_weighting))
-            .add(highway_rest_area.multiply(self.highway_rest_area_weighting))
-            .add(highway_road.multiply(self.highway_road_weighting))
-            .add(highway_secondary.multiply(self.highway_secondary_weighting))
-            .add(highway_secondary_link.multiply(self.highway_secondary_link_weighting))
-            .add(highway_service.multiply(self.highway_service_weighting))
-            .add(highway_steps.multiply(self.highway_steps_weighting))
-            .add(highway_tertiary.multiply(self.highway_tertiary_weighting))
-            .add(highway_tertiary_link.multiply(self.highway_tertiary_link_weighting))
-            .add(highway_track.multiply(self.highway_track_weighting))
-            .add(highway_trunk.multiply(self.highway_trunk_weighting))
-            .add(highway_trunk_link.multiply(self.highway_trunk_link_weighting))
-            .add(highway_turning_circle.multiply(self.highway_turning_circle_weighting))
-            .add(highway_unclassified.multiply(self.highway_unclassified_weighting))
-            .add(ee.Image(self.inputs["groads_additions"]["ee_path"]).multiply(self.highway_groads_weighting))
-        ).multiply(2)
-
-
-        roads_bool = highway_total.gt(0)
+    def calc(self):
+        if self.osm:
+            self.osm_groads_combined_influence()
+        else:
+            self.groads_influence()
 
-        roads_indirect = roads_bool.eq(0)\
-            .cumulativeCost(roads_bool, 15000)\
-            .reproject(crs=self.crs, scale=self.scale)\
-            .multiply(self.DECAY_CONSTANT)\
-            .exp()\
-            .multiply(self.INDIRECT_INFLUENCE)\
+        road_driver = (
+            self.road_direct_cost.addBands(self.road_indirect_cost)
+            .reduce(ee.Reducer.max())
             .unmask(0)
+            .updateMask(self.water)
+            .multiply(100)
+            .int()
+            .rename("hii_road_driver")
+        )
 
-        roads_total = highway_total.add(roads_indirect).updateMask(watermask)
+        # # TODO: determine if drivers are to be normalized or not.
```

Comment:
I think we should go ahead with this; we effectively already are. As discussed, I think if we just set another class constant up top about what the top value should be then this should be straightforward; we could make it 8 like Venter and then add weights as has been done previously, or make it 0-100 (my preference) and then in the add_weights task weight the drivers in a way that's equivalent to the relative importance assigned previously.

Reply:
I pushed the current version. might be good to get what we have merged, and develop this logic on a new branch